src/createXcmTypes/util/fetchPalletInstanceId.ts

```diff
@@ -11,6 +11,10 @@ import { BaseError, BaseErrorsEnum } from '../../errors';
  * @param isLiquidToken Boolean to determine whether or not to fetch the PoolAssets id.
  */
 export const fetchPalletInstanceId = (api: ApiPromise, isLiquidToken: boolean, isForeignAsset: boolean): string => {
+	if (!api.query.assets) {
```

Comment:
DQ: But if the assets pallet is not available than we can assume that `PoolAssets`, and `ForeignAssets` will be unavailable as well?

Reply:
thats a great point. As is we would not check whether it is a pool assets or foreign assets tx if the assets pallet is not supported. I've updated this in order to evaluate the `poolAssets` and `foreignAssets` pallet cases first. We will only return an empty string prior to filtering the pallet names if the assets pallet isn't available and the palletName is an empty string.

Reply:
Sounds perfect! Thanks for the response. Looks like a solid change!